com.unity.netcode.gameobjects/Tests/Runtime/DistributedAuthority/SessionVersionConnectionRequest.cs

```diff
@@ -0,0 +1,106 @@
+
+        private bool m_UseValidSessionVersion;
+        private bool m_ClientWasDisconnected;
+        private NetworkManager m_ClientNetworkManager;
+
+        /// <summary>
+        /// Callback used to mock the scenario where a client has an invalid session version
+        /// </summary>
+        /// <returns><see cref="SessionConfig"/></returns>
+        private SessionConfig GetInavlidSessionConfig()
+        {
+            return new SessionConfig(m_ServerNetworkManager.SessionConfig.SessionVersion - 1);
+        }
+
+        /// <summary>
+        /// Overriding this method allows us to configure the newly instantiated client's
+        /// NetworkManager prior to it being started.
+        /// </summary>
+        /// <param name="networkManager">the newly instantiated NetworkManager</param>
+        protected override void OnNewClientCreated(NetworkManager networkManager)
+        {
+            m_ClientWasDisconnected = false;
+            m_ClientNetworkManager = networkManager;
+            m_ClientNetworkManager.OnClientDisconnectCallback += OnClientDisconnectCallback;
+            if (!m_UseValidSessionVersion)
+            {
+                networkManager.OnGetSessionConfig = GetInavlidSessionConfig;
+            }
+            base.OnNewClientCreated(networkManager);
+        }
+
+        /// <summary>
+        /// Tracks if the client was disconnected or not
+        /// </summary>
+        private void OnClientDisconnectCallback(ulong clientId)
+        {
+            m_ClientWasDisconnected = true;
+            m_ClientNetworkManager.OnClientDisconnectCallback -= OnClientDisconnectCallback;
+        }
+
+        /// <summary>
+        /// This handles disabling the internal integration test logic that waits for
+        /// clients to be connected. When we know the client will be disconnected,
+        /// we want to have the NetcodeIntegrationTest not wait for the client to
+        /// connect (otherwise it will timeout there and fail the test).
+        /// </summary>
+        /// <param name="networkManager"></param>
+        /// <returns>true = wait | false = don't wait</returns>
+        protected override bool ShouldWaitForNewClientToConnect(NetworkManager networkManager)
+        {
+            return m_UseValidSessionVersion;
+        }
+
+        /// <summary>
+        /// Validates that when the client's session config version is valid a client will be
+        /// allowed to connect and when it is not valid the client will be disconnected.
+        /// </summary>
+        /// <remarks>
+        /// This is just a mock of the service logic to validate everything on the NGO side is
+        /// working correctly.
+        /// </remarks>
+        /// <param name="useValidSessionVersion">true = use valid session version | false = use invalid session version</param>
+        [UnityTest]
+        public IEnumerator ValidateSessionVersion([Values] bool useValidSessionVersion)
```

Comment:
Out of interest, what is the benefit of combining both tests, rather than having two separate tests?

Reply:
Nothing really... it is a matter of preference in this particular case. Using TestFixtures and Values helps to reduce the script footprint... but in reality it is a matter of preference...
I personally prefer to have a test validating a specific feature/function of NGO to be under one test as opposed to several to avoid replicating code.